addon/ajax-request.js

```diff
@@ -148,6 +164,17 @@ export default class AjaxRequest {
   }
 
   /**
+   * @method _getFullHeadersHash
+   * @private
+   * @param {Object} headers
+   * @return {Object}
+   */
+  _getFullHeadersHash(headers) {
+    const classHeaders = get(this, 'headers') || {};
+    return merge(classHeaders, headers);
```

Comment:
when headers are set on the service, this operation will mutate that headers object. This is likely not the desired behaviour.

You might want to try something like:
```
return {...classHeaders, ...headers};
```